File: plugins/custom_native_video_player_ios/android/src/main/kotlin/me/albemala/native_video_player/NativeVideoPlayerViewFactory.kt
package me.albemala.native_video_player

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeVideoPlayerViewFactory(
    private val messenger: BinaryMessenger
) :
    PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    companion object {
        const val id = "native_video_player_view"
    }

    override fun create(
        context: Context?,
        viewId: Int,
        args: Any?
    ): PlatformView {
        return NativeVideoPlayerViewController(
            messenger,
            viewId,
            context
        )
    }
}

File: plugins/custom_native_video_player_ios/ios/Classes/NativeVideoPlayerViewFactory.swift
import Foundation
import Flutter

public class NativeVideoPlayerViewFactory: NSObject, FlutterPlatformViewFactory {
    public static let id = "native_video_player_view"

    private let messenger: FlutterBinaryMessenger

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
    }

    public func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        NativeVideoPlayerViewController(
            messenger: messenger,
            viewId: viewId,
            frame: frame
        )
    }
}
